ecopod/sendRawTransactionConditional.md

```diff
@@ -0,0 +1,88 @@
+
+If high rates of rejected transactions are observed, additional validation rules & improvements can be iterated on.
+
+* **Local Rate Limiting**
+
+    Adopt flashbots-style [reputation](https://docs.flashbots.net/flashbots-auction/advanced/reputation) score that's computed based on the success rate of conditional txs. Designed to filter out inefficient callers and promote honest ones.
+
+* **Optimized State Lookups**
+
+    Verifying the conditional associated with the transaction requires state access, for which the sequencer is not compensated when the transaction is rejected. To minimize resource usage, these checks could also be applied by the validating proxy prior to the sequencer mempool. This could then leverage an optimized state store through solutions like [revm](https://github.com/bluealloy/revm) to reject failed conditionals both cheaper and faster.
+
+# Risks & Uncertainties
+
+**Risk 1: Private 4337 mempools remain the status quo.** Bundlers can still suffer from reverts and have a strong desire for an endpoint like `eth_sendRawTransactionConditional`. Even without shared 4337 mempools, this endpoint adds value bundlers. In our proposed solution, we can allowlist these bundlers and hold them to an SLA. It also provides the opportunity for chain operators to verticalize their 4337 offering through the same allowlist functionality.
+
+**Risk 2: Implemented validation isn't enough for permissionless bundler participation.** The listed validation rules are a starting point and there's room for exploration in horizontally scalable validation. However we can fallback to a permissioned allowlist for this endpoint which still enables 4337 shared mempools and likely makes no difference to dapp developers which already use a small subset of known infra providers.
+
+**Risk 3: Generalized External Validation.** Validation policies should be DRY'd between interop, eth_sendRawTransactionConditional, and any future use cases. These policies that are implementated should work well between these usecases as this approach is adopted and scales. The tech-debt here can grow quickly if each solution has it's own methods of preventing DoS and validation, especially operationally.
+
+**Risk 4: Excessive Compute/Operational Requirements**. This endpoint is a feature provided out of protocol by the block builder -- the sequencer. With failed conditional transactions, the sequencer is not compensated with charged gas like when processing a reverted transaction. There's also the added overhead of managing new services that must mitigate DoS and increases the surface area where manual intervention will be required. The wasted compute on failed conditionals or inability to effectively mitigate DoS may be a reason to rollback this feature.
```

Comment:
Thanks for including this section :)

Just want to note that even when the transaction is included (conditional passes), the compute to check the invariant is still unpaid. And in fact, compute pressure from a passing invariant is higher than compute pressure from a failed one (because passing invariants are checked on every recommit, while a failed one would be dropped and not recomputed)